app/entrypoint.sh

```diff
@@ -58,7 +58,11 @@ if [ "$1" = 'platform' ]; then
   if [ -z "$MM_SQLSETTINGS_DATASOURCE" ]
   then
     echo -ne "Configure database connection..."
-    export MM_SQLSETTINGS_DATASOURCE="postgres://$MM_USERNAME:$MM_PASSWORD@$DB_HOST:$DB_PORT_NUMBER/$MM_DBNAME?sslmode=disable&connect_timeout=10"
+    # URLEncode the password, allowing for special characters.
+    if [ "${MM_PASSWORD}" ]; then
+        ENCODED_PASSWORD=$(python -c "import urllib; print urllib.quote('''$MM_PASSWORD''')")
+    fi
```

Comment:
Since we use `jq`, you can do this more simply with it: `printf %s $MM_PASSWORD | jq -s -R -r @uri`

And that'll work on Alpine as well.